lib/ansible/modules/cloud/amazon/ec2_vpc_igw_facts.py

```diff
@@ -0,0 +1,167 @@
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.ec2 import boto3_conn, camel_dict_to_snake_dict
+
+
+DOCUMENTATION = '''
+---
+module: ec2_vpc_igw_facts
+short_description: Gather facts about internet gateways in AWS
+description:
+    - Gather facts about internet gateways in AWS
+version_added: "2.3"
+author: "Nick Aslanidis (@naslanidis)"
+options:
+  filters:
+    description:
+      - A dict of filters to apply. Each dict item consists of a filter key and a filter value.
+         See U(http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeRouteTables.html) for possible filters.
+    required: false
+    default: null
+  InternetGatewayIds:
+    description:
+      - Get details of specific Internet Gateway ID
+      - Provide this value as a list
+    required: false
+    default: None
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# # Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Gather facts about all Internet Gateways for an account or profile
+  ec2_vpc_igw_facts:
+    region: ap-southeast-2
+    profile: production
+  register: igw_facts
+
+- name: Gather facts about a filtered list of Internet Gateways
+  ec2_vpc_igw_facts:
+    region: ap-southeast-2
+    profile: production
+    filters:
+        "tag:Name": "igw-123"
+  register: igw_facts
+
+- name: Gather facts about a specific internet gateway by InternetGatewayId
+  ec2_vpc_igw_facts:
+    region: ap-southeast-2
+    profile: production
+    InternetGatewayIds: igw-c123f6a7
+  register: igw_facts
+'''
+
+RETURN = '''
+internet_gateways:
+    description: The internet gateways for the account
+    returned: always
+    type: list
+
+changed:
+    description: True if listing the internet gateways succeeds
+    type: bool
+    returned: always
+'''
+
+try:
+    import botocore
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_internet_gateway_info(internet_gateway):
+    internet_gateway_info = {'InternetGatewayId': internet_gateway['InternetGatewayId'],
+                             'Attachments': internet_gateway['Attachments'],
+                             'Tags': internet_gateway['Tags'] }
+    return internet_gateway_info
+
+
+def list_internet_gateways(client, module):
+    all_internet_gateways_array = []
+    params = dict()
+
+    if module.params.get('filters'):
+        params['Filters'] = []
+        for key, value in module.params.get('filters').items():
+            temp_dict = dict()
+            temp_dict['Name'] = key
+            if isinstance(value, basestring):
+                temp_dict['Values'] = [value]
+            else:
+                temp_dict['Values'] = value
+            params['Filters'].append(temp_dict)
+
+    dryrun = module.params.get("DryRun")
```

Comment:
set `DryRun` default to `False` and then `params['DryRun'] = module.params.get('dryrun')`